File: server/src/main/com/broll/gainea/server/core/cards/impl/play/C_TeleportMountain.java
package com.broll.gainea.server.core.cards.impl.play;

import com.broll.gainea.server.core.cards.TeleportCard;
import com.broll.gainea.server.core.map.AreaType;
import com.broll.gainea.server.core.map.Location;
import com.broll.gainea.server.core.utils.LocationUtils;

import java.util.List;
import java.util.stream.Collectors;

public class C_TeleportMountain extends TeleportCard {
    public C_TeleportMountain() {
        super(44, "Bergwanderung", "Bewegt eine Truppe zu einem beliebigen Berg auf der gleichen Karte");
    }

    @Override
    public List<Location> getTeleportTargets(Location from) {
        return LocationUtils.filterByType(from.getContainer().getExpansion().getAllLocations(), AreaType.MOUNTAIN).collect(Collectors.toList());
    }
}

File: server/src/main/com/broll/gainea/server/core/cards/impl/play/C_TeleportMountain.kt
package com.broll.gainea.server.core.cards.impl.play

import com.broll.gainea.server.core.Game
import com.broll.gainea.server.core.cards.TeleportCard
import com.broll.gainea.server.core.map.AreaType.MOUNTAIN
import com.broll.gainea.server.core.map.Location

class C_TeleportMountain : TeleportCard(44, "Bergwanderung", "Bewegt eine Truppe zu einem beliebigen Berg auf der gleichen Karte") {
    override fun getTeleportTargets(from: Location) = from.container.expansion.allAreas.filter { it.type == MOUNTAIN }

    override fun validFor(game: Game) = game.map.allAreas.any { it.type == MOUNTAIN }
}
